### PR TITLE
Update e2e tests to use automatic platform detection

### PR DESCRIPTION
**Pull request by gkrenn**
# Description

Removes the dependency of the e2e tests on the platform env var.
Openshift is automatically detected by checking if SCCs group version is available in the cluster.

## How can this be tested?

Run e2e tests in openshift and kubernetes.

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/kubeobjects/platform.go [hunk 1]
```diff
@@ -15,22 +20,54 @@ const (
...
 
-func ResolvePlatformFromEnv() Platform {
-	switch os.Getenv(platformEnvName) {
-	case openshiftPlatformEnvValue:
-		return Openshift
-	case kubernetesPlatformEnvValue:
-		fallthrough
-	default:
-		return Kubernetes
+const SccGVR = "security.openshift.io/v1"
```
Comment (0sewa0) [inline src/kubeobjects/platform.go:23]: 😭 🤮  (I don't have a better name, `OpenshiftSecurityGVR`, but its not much better)

File: src/kubeobjects/platform.go [hunk 2]
```diff
@@ -15,22 +20,54 @@ const (
...
+
+type discoveryClientCreation interface {
```
Comment (0sewa0) [inline src/kubeobjects/platform.go:25]:
```suggestion
type discoveryClientProvider interface {
```

File: src/kubeobjects/platform.go [hunk 3]
```diff
@@ -15,22 +20,54 @@ const (
...
+	getDiscoveryClient() (discovery.DiscoveryInterface, error)
+}
+
+type DiscoveryClientResolver struct{}
+
+func (p *DiscoveryClientResolver) getDiscoveryClient() (discovery.DiscoveryInterface, error) {
```
Comment (0sewa0) [inline src/kubeobjects/platform.go:31]: THERE IS NOT EVEN THE LETTER `p` IN THE TYPE NAME 😂 
```suggestion
func (resolver *DiscoveryClientResolver) getDiscoveryClient() (discovery.DiscoveryInterface, error) {
```

**Review by 0sewa0 — CHANGES_REQUESTED** [review]

File: src/kubeobjects/platform.go (hunk 1 above)
Comment (StefanHauth) [inline src/kubeobjects/platform.go:23]: Does this have to be public?

File: test/scenarios/cloudnative/proxy/proxy.go [hunk 4]
```diff
@@ -60,7 +60,7 @@ func withProxy(t *testing.T, proxySpec *dynatracev1beta1.DynaKubeProxy) features
 	assess.InstallOperatorFromSourceWithCustomNamespace(builder, operatorNamespaceBuilder.Build(), testDynakube)
 
 	// Register proxy create and delete
-	proxy.SetupProxyWithTeardown(builder, testDynakube)
+	proxy.SetupProxyWithTeardown(builder, t, testDynakube)
```
Comment (StefanHauth) [inline test/scenarios/cloudnative/proxy/proxy.go:63]: Matter of taste: testing.T should always be the first paramter.

File: src/kubeobjects/platform.go [hunk 5]
```diff
@@ -15,22 +20,54 @@ const (
...
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+type PlatformResolver struct {
+	discoveryClientCreation
+}
```
[the remark this answers is not in the document]
Comment (gkrenn) [reply src/kubeobjects/platform.go:48]: Will do

File: src/kubeobjects/platform.go [hunk 6]
```diff
@@ -1,9 +1,14 @@
 package kubeobjects
```
Comment (0sewa0) [inline src/kubeobjects/platform.go:1]: there is a `kubeobjects` under `test` (for example: https://github.com/Dynatrace/dynatrace-operator/blob/03695c1d95fe519ea50492906016410452ed8926/test/helpers/kubeobjects/environment/environment.go#L14)

as these functions are only used for testing we should have it under `/test`

**Review by 0sewa0 — CHANGES_REQUESTED** [review]

**Review by 0sewa0 — APPROVED** [review]

**Pull request merged**